sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -103,9 +105,11 @@ class SQLContext(@transient val sparkContext: SparkContext)
    */
   def getAllConfs: immutable.Map[String, String] = conf.getAllConfs
 
+  // TODO how to handle the temp table per user session?
   @transient
   protected[sql] lazy val catalog: Catalog = new SimpleCatalog(true)
 
+  // TODO how to handle the temp function per user session?
```

Comment:
Yea, the same with `Catalog`, we also need to think about the how to handle the `temp` function for current session in `FunctionRegistry`.